src/test/scala/vsys/blockchain/state/opcdiffs/BasicOpcDiffTest.scala

```diff
@@ -132,6 +209,20 @@ class BasicOpcDiffTest extends PropSpec with PropertyChecks with GeneratorDriven
       DataEntry(Longs.toByteArray(1), DataType.ShortText),
       Seq.empty, 0, BasicOpcDiff.maximum) should be (Left(ContractUnsupportedOPC))
 
+    BasicOpcDiff.operation(
+      DataEntry(Shorts.toByteArray(BigInt(2).toByteArray.length.toShort) ++ BigInt(2).toByteArray, DataType.BigInteger),
+      DataEntry(Shorts.toByteArray(BigInt(1).toByteArray.length.toShort) ++ BigInt(1).toByteArray, DataType.BigInteger),
+      Seq.empty, 0, BasicOpcDiff.maximum) should be (Right(Seq(DataEntry(
+      Shorts.toByteArray(BigInt(2).toByteArray.length.toShort) ++ BigInt(2).toByteArray, DataType.BigInteger))))
+    BasicOpcDiff.operation(
+      DataEntry(Shorts.toByteArray(BigInt(0).toByteArray.length.toShort) ++ BigInt(0).toByteArray, DataType.BigInteger),
+      DataEntry(Shorts.toByteArray(BigInt(0).toByteArray.length.toShort) ++ BigInt(0).toByteArray, DataType.BigInteger),
+      Seq.empty, 1, BasicOpcDiff.maximum) should be (Left(ContractLocalVariableIndexOutOfRange))
+    BasicOpcDiff.operation(
+      DataEntry(Shorts.toByteArray(BigInt(0).toByteArray.length.toShort) ++ BigInt(0).toByteArray, DataType.BigInteger),
+      DataEntry(Longs.toByteArray(1), DataType.Timestamp),
+      Seq.empty, 0, BasicOpcDiff.maximum) should be (Left(ContractDataTypeMismatch))
+
```

Comment:
@Icermli an sqrt method is added for BigInteger DataEntry, test it too

Reply:
ok